runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -502,7 +502,17 @@ static void setupAvailableParallelism(int32_t maxThreads) {
     if (hwpar > 0) {
         // Limit the parallelism to the maximum imposed by the comm layer.
         if (0 < maxThreads && maxThreads < hwpar) {
-            hwpar = maxThreads;
+          if (chpl_nodeID == 0) {
+            char msg[1024];
+            snprintf(msg, sizeof(msg),
+                     "The CHPL_COMM setting is limiting the number of threads "
+                     "to %d, rather than the hardware's preference of %d.  "
+                     "If using CHPL_COMM=gasnet, set 'GASNET_MAX_THREADS' to "
```

Comment:
I think this would be better if it just check against `CHPL_COMM` to specialize the error message for GASNet. But my understanding of this PR is that in practice this warning can only occur with GASNet, so I say leave it unless we see it occur on non-gasnet systems

Reply:
>  in practice this warning can only occur with GASNet

I believe that's accurate for today's backends.

Reply:
> I think this would be better if it just checks against CHPL_COMM

I agree that'd be better.  In putting this together very rapidly, I couldn't remember how to check `CHPL_COMM` within the runtime, nor how easy it was, and was apparently too lazy to look.  Now that Jade's reminded me, I'm going to tighten this up, less because I think it'll ever happen in practice (well, in the near-term anyway) and more because I don't really like the "If…" conditional in the warning when we really ought to know the answer.

Reply:
Improved this in https://github.com/chapel-lang/chapel/pull/26448/commits/d44edd2a8559f719ffa5789260a4f147bc6d66ee and caught/fixed a longstanding bug in testing it.